src/modules/package-index.ts

```diff
@@ -12,185 +12,200 @@
- * @prop {string} language Programming language the example is available.
+ * Represents a set of parameters with an ID, name, and an array of parameters.
+ * @interface ParameterSet
+ * @prop {string} id - Unique ID of the parameter set.
+ * @prop {string} name - Name of the parameter set.
+ * @prop {Parameter[]} parameters - Array of parameters in the set.
  */
-export interface ExampleDescription {
+export interface ParameterSet {
+    id: string;
     name: string;
-    value: string;
-    language: string;
+    parameters: Parameter[];
 }
 
 /**
- * Additional argument for exposed interface
- * @interface Argument
- * @prop {string} id Unique ID of an argument for an exposed interface.
- * @prop {string} description Description of the argument.
- * @prop {string} type Datatype of the argument.
- * @prop {string} [default] Default value of the argument.
- * @prop {boolean} [required] Determines if the argument is required.
+ * Exposed interface of a package, which can be either a Core or an Extension.
+ * @interface ExposedInterface
+ * @prop {string} type - Type (Name/ID) of the exposed interface.
+ * @prop {string} description - Description of the exposed interface.
+ * @prop {Core | Extension} args - Additional arguments an exposed interface needs.
  */
-export interface Argument {
+type ExposedInterface = Core | Extension;
+
+/**
+ * Represents the core part of an exposed interface with an ID, type, name, description, and optional parameter sets.
+ * @interface Core
+ * @prop {string} id - Unique ID of the core.
+ * @prop {string} type - Type of the core.
+ * @prop {string} name - Name of the core.
+ * @prop {string} description - Description of the core.
+ * @prop {ParameterSet[]} [parameterSets] - Array of parameter sets.
+ */
+export interface Core {
     id: string;
-    description: string;
     type: string;
-    default?: string;
-    required?: boolean;
-    [key: string]: any;
+    name: string;
+    description: string;
+    parameterSets?: ParameterSet[];
 }
 
 /**
- * Exposed interface of a package
- * @interface ExposedInterface
- * @prop {string} type Name/ID of the exposed interface.
- * @prop {string} description Description of the exposed interface.
- * @prop {Argument[]} args Additional arguments an exposed interface need.
- * @prop {boolean} [default] Determines if the exposed interface is used as default.
+ * Represents the extension part of an exposed interface with an ID, type, name, description, compatible cores, and optional parameters.
+ * @interface Extension
+ * @prop {string} id - Unique ID of the extension.
+ * @prop {string} type - Type of the extension.
+ * @prop {string} name - Name of the extension.
+ * @prop {string} description - Description of the extension.
+ * @prop {string[]} compatibleCores - Array of compatible core types.
+ * @prop {Parameter[]} [parameters] - Array of parameters.
  */
-export interface ExposedInterface {
+export interface Extension {
```

Comment:
Introduce a common base interface for Core and Extension which has id, type, name and description attributes since they both share that - that way you can also remove the defined `ExposedInterface` type since you can use the base interface instead.